### PR TITLE
FISH-5748 Added Environment variable functionality

### PR DESCRIPTION
**Pull request by kalinchan**
Created a function to get the System environment variables. 

Tested with openid application in payara-samples/samples/openid by changing OpenIdAuthenticationDefinition to the following:
```
        providerURI = "http://localhost:8080/openid-server/webresources/oidc-provider",
        clientId = "${ENV=OIDC_CLIENTID}",
        clientSecret = "${ENV=OIDC_CLIENTSECRET}",
        redirectURI = "${baseURL}/Callback"
```

### CODE REVIEW

**Comment by Pandrex247** [comment]
CC @pdudits & @jGauravGupta as the experts here.

File: openid/src/main/java/fish/payara/security/openid/controller/ConfigurationController.java [hunk 1]
```diff
@@ -494,4 +488,19 @@ static CacheKey keyFromDefinition(OpenIdAuthenticationDefinition definition) {
         } : new String[5];
     }
 
+    private String getEnvironmentProperty(String property) {
```
Comment (pdudits) [inline openid/src/main/java/fish/payara/security/openid/controller/ConfigurationController.java:491]: If I put in clientSecret of `b78d6xPATH`, should I get that replaced with value of PATH environment variable? I suppose not.

**Review by pdudits — CHANGES_REQUESTED** [review]

**Comment by pdudits** [comment]
> I don't think this is complete. You're only supporting ENV variables for clientSecret and clientId, but what about all the other options available for the OpenIdAuthenticationDefinition annotation?

Since all of those are controlled by MP Config properties, even the ones covered here can already be controlled by env variables `PAYARA_SECURITY_OPENID_CLIENTID` and `PAYARA_SECURITY_OPENID_CLIENTSECRET`.

> Also I think the issue scope is too limited - by the looks of it, everything that we previously supported via going through [TranslatedConfigView#getTranslatedValue](https://github.com/payara/Payara/blob/master/nucleus/admin/config-api/src/main/java/org/glassfish/config/support/TranslatedConfigView.java#L119) has been ripped out (password aliases, env properties, system properties...). Related docs: https://docs.payara.fish/community/docs/documentation/payara-server/public-api/openid-connect-support.html#secret-alias

Being outside of main codebase means that we cannot access internal server APIs anymore (until they become separate repository we can have dependency onto).

That was the reason for having [payara-expression-config-properties](https://docs.payara.fish/community/docs/documentation/payara-server/server-configuration/var-substitution/usage-of-variables.html#examples-ver-ref): so that TranslatedConfigview is hidden behind standardized interface. That would be second way of configuring all of these with environment variables and everything else.

Finally, there are [MP config expressions](https://github.com/eclipse/microprofile-config/blob/master/spec/src/main/asciidoc/property-expressions.asciidoc) that allow to compose property values as well and standardized mapping to env variables exist. That's third way of configuring those via environment property.

I don't understand why we need a fourth way.

**Comment by kalinchan** [comment]
Since there are alternative ways, this will be closed and Documentation will be updated.

**Pull request closed without merging**